src/jit/codegenarm64.cpp

```diff
@@ -2470,7 +2470,8 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                 // We unconditionally throw a divide by zero exception
                 genJumpToThrowHlpBlk(EJ_jmp, SCK_DIV_BY_ZERO);
 
-                // We still need to call genProduceReg
+                // We still need to call genConsumeOperands/genProduceReg for liveness update.
+                genConsumeOperands(treeNode->AsOp());
```

Comment:
@sivarv I don't think it's the right thing to make all the callers of genCodeForBinary() make the explicit call the genConsumeOperands(). It would be better to add a method, to be called by genCodeForBinary() as well as this, that does the work required by both it and the more complex case.